problems/arraysWithLoops.js

```diff
@@ -25,39 +38,71 @@ function shoutWhileLoop() {}
  * @returns {Array} - ["I!", "am!", "a!", "happy!", "array!"]
  */
 
-function shoutForOfLoop() {}
+const shoutForOfLoop = (arr) => {
+  let newArray = []
+  for(let element of arr){
+    newArray.push(element + "!");
+  }
+return newArray;
+}
+  
 
 /**
  * Takes in an array and returns the sum of all values
  * @param {number[]} nums
  * @returns {number} sum
  */
 
-function sumArray() {}
+const sumArray = (nums) => {
+  let sum = 0; 
+        for(let i = 0; i < nums.length; i++ ) {
+            sum += nums[i];
+        }    
+    return sum;
+}
 
 /**
  * Takes in an array and returns an array of all the odd valued elements
  * @param {number[]} nums
  * @returns {array} odds
  */
 
-function oddArray() {}
+const oddArray = (nums) => {
+  let odds = []
+  for (let i = 0; i < nums.length; i++) {
+    if (nums[i] % 2 === 0) continue;
+    odds.push(nums[i]);
+  }
+  return odds
+}
 
 /**
  * Takes in an array and returns an array of all the even valued elements
  * @param {number[]} nums
  * @returns {array} evens
  */
 
-function evenArray() {}
-
+const evenArray = (nums)=> {
+  let evens = []
+  for (let i = 0; i < nums.length; i++) {
+    if (nums[i] % 2 === 1) continue;
+    evens.push(nums[i]);
+  }
+  return evens
+}
 /**
  * Takes in array and returns the smallest number in the array
  * @param {number[]} nums
  * @returns {number} smallest value
  */
 
-function findSmallest() {}
+const findSmallest = (nums) => {
+  let smallest = Math.min.apply(Math, nums);
+  nums.indexOf(smallest), 1
```

Comment:
This lines not doing anything. 
Could just do return Math.min(...nums)
But point was to iterate through the array.